### PR TITLE
Code improvements for Gnuton PR.

### PR DESCRIPTION
**Pull request by Martinski4GitHub**
Some code improvements & fine-tuning for the newly merged code that supports Gnuton F/W updates.

### CODE REVIEW

**Comment by ExtremeFiretop** [comment]
@Martinski4GitHub 

This is all fantastic! Great great stuff!
I'll start my review right away!

**Comment by Martinski4GitHub** [comment]
> @Martinski4GitHub
> 
> This is all fantastic! Great great stuff! I'll start my review right away! Thank you again!
>
As you said, we had to "bite the bullet" at some point and merge the Gnuton PR into the development branch.
I'll check back in the evening. Have a good Monday, bud!!

File: MerlinAU.sh [hunk 1]
```diff
@@ -5291,10 +5311,12 @@ _RunBackupmon_()
 
             # Adjust version format from 1.46 to 1.4.6 if needed
             local DOT_COUNT="$(echo "$BM_VERSION" | tr -cd '.' | wc -c)"
-            if [ "$DOT_COUNT" -eq 0 ]; then
+            if [ "$DOT_COUNT" -eq 0 ]
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:5314]: Approved

File: MerlinAU.sh [hunk 2]
```diff
@@ -5291,10 +5311,12 @@ _RunBackupmon_()
...
+            then
                 # If there's no dot, it's a simple version like "1" (unlikely but let's handle it)
                 BM_VERSION="${BM_VERSION}.0.0"
-            elif [ "$DOT_COUNT" -eq 1 ]; then
+            elif [ "$DOT_COUNT" -eq 1 ]
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:5318]: Approved

File: MerlinAU.sh [hunk 3]
```diff
@@ -5304,15 +5326,17 @@ _RunBackupmon_()
             local requiredBM_version="$(_ScriptVersionStrToNum_ "1.5.3")"
 
             # Check if BACKUPMON version is greater than or equal to 1.5.3
-            if [ "$currentBM_version" -ge "$requiredBM_version" ]; then
+            if [ "$currentBM_version" -ge "$requiredBM_version" ]
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:5329]: Approved

File: MerlinAU.sh [hunk 4]
```diff
@@ -5304,15 +5326,17 @@ _RunBackupmon_()
...
+            then
                 # Execute the backup script if it exists #
                 echo ""
                 Say "Backup Started (by BACKUPMON)"
                 sh /jffs/scripts/backupmon.sh -backup >/dev/null
                 BE=$?
                 Say "Backup Finished"
                 echo ""
-                if [ $BE -eq 0 ]; then
+                if [ $BE -eq 0 ]
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:5338]: Approved

File: MerlinAU.sh [hunk 5]
```diff
@@ -618,12 +607,14 @@ readonly FW_UpdateNotificationDateFormat="%Y-%m-%d_%H:%M:00"
...
+
 ##FOR TESTING/DEBUG ONLY##
-#readonly PRODUCT_ID="TUF-AX3000_V2"
+##readonly PRODUCT_ID="TUF-AX3000_V2"
 ##FOR TESTING/DEBUG ONLY##
+
 readonly FW_FileName="${PRODUCT_ID}_firmware"
 readonly FW_SFURL_RELEASE="${FW_SFURL_BASE}/${PRODUCT_ID}/${FW_SFURL_RELEASE_SUFFIX}/"
-readonly isGNUtonFW=$(_GetFirmwareVariantFromRouter_)  
+readonly isGNUtonFW="$(_GetFirmwareVariantFromRouter_)"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:617]: Approved

File: MerlinAU.sh [hunk 6]
```diff
@@ -5140,10 +5147,14 @@ _SelectOfflineUpdateFile_()
 
         # Prompt user to select a file #
         printf "\n---------------------------------------------------\n"
-        if "$isGNUtonFW"; then printf "\n[${theMUExitStr}] Enter the number of the update file you want to select:  "
-        else printf "\n[${theMUExitStr}] Enter the number of the ZIP file you want to select:  "; fi
-        read -r selection
+        if "$isGNUtonFW"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:5150]: Approved

File: MerlinAU.sh [hunk 7]
```diff
@@ -5126,11 +5130,14 @@ _SelectOfflineUpdateFile_()
 
     while true
     do
-        if "$isGNUtonFW"; then printf "\nAvailable update files in the directory: [${GRNct}${FW_ZIP_DIR}${NOct}]:\n\n"
-        else printf "\nAvailable ZIP files in the directory: [${GRNct}${FW_ZIP_DIR}${NOct}]:\n\n"; fi
-
-        if "$isGNUtonFW"; then fileList="$(ls -A "$FW_ZIP_DIR"/*.w "$FW_ZIP_DIR"/*.pkgtb 2>/dev/null)"
-        else fileList="$(ls -A "$FW_ZIP_DIR"/*.zip 2>/dev/null)"; fi
+        if "$isGNUtonFW"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:5133]: Approved

**Review by ExtremeFiretop — APPROVED** [review]

File: MerlinAU.sh [hunk 8]
```diff
@@ -4584,10 +4584,10 @@ _ManageChangelogMerlin_()
     wget --timeout=5 --tries=4 --waitretry=5 --retry-connrefused \
          -O "$changeLogFile" -o "$wgetLogFile" "${changeLogURL}"
 
-    if [ ! -f "$changeLogFile" ]
+    if [ ! -s "$changeLogFile" ]
     then
         Say "Changelog file [$changeLogFile] does NOT exist."
-        echo ; [ -f "$wgetLogFile" ] && cat "$wgetLogFile"
+        echo ; [ -s "$wgetLogFile" ] && cat "$wgetLogFile"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:4590]: Approved

File: MerlinAU.sh [hunk 9]
```diff
@@ -4432,7 +4435,8 @@ _ChangelogVerificationCheck_()
 
         if "$isGNUtonFW"
         then
-            changeLogFile="${FW_BIN_DIR}/${FW_FileName}_Changelog.txt"
+            changeLogFName="${FW_FileName}_Changelog.txt"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:4438]: Approved

File: MerlinAU.sh [hunk 10]
```diff
@@ -4444,17 +4448,13 @@ _ChangelogVerificationCheck_()
             else
                 changeLogTag="NG"
             fi
-            changeLogFile="$(/usr/bin/find -L "${FW_BIN_DIR}" -name "Changelog-${changeLogTag}.txt" -print)"
+            changeLogFName="Changelog-${changeLogTag}.txt"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:4451]: Approved

File: MerlinAU.sh [hunk 11]
```diff
@@ -4444,17 +4448,13 @@ _ChangelogVerificationCheck_()
...
+            changeLogFPath="$(/usr/bin/find -L "${FW_BIN_DIR}" -name "$changeLogFName" -print)"
         fi
 
-        if [ ! -f "$changeLogFile" ]
+        if [ ! -f "$changeLogFPath" ]
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:4455]: Approved

File: MerlinAU.sh [hunk 12]
```diff
@@ -4444,17 +4448,13 @@ _ChangelogVerificationCheck_()
...
         then
-            if "$isGNUtonFW"
-            then
-                Say "Changelog file [${FW_BIN_DIR}/${FW_FileName}_Changelog.txt] does NOT exist."
-            else
-                Say "Changelog file [${FW_BIN_DIR}/Changelog-${changeLogTag}.txt] does NOT exist."
-            fi
+            Say "Changelog file [${FW_BIN_DIR}/${changeLogFName}] does NOT exist."
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:4457]: Approved

File: MerlinAU.sh [hunk 13]
```diff
@@ -4498,15 +4498,15 @@ _ChangelogVerificationCheck_()
...
+                if ! grep -Eq "$current_version_regex" "$changeLogFPath"
+                then
                     Say "Current version NOT found in changelog file. Bypassing changelog verification for this run."
                     return 0
-                else
-                    # Extract log contents between two firmware versions for non-Gnuton changelogs #
-                    changelog_contents="$(awk "/$release_version_regex/,/$current_version_regex/" "$changeLogFile")"
                 fi
+                # Extract log contents between two firmware versions from RMerlin #
+                changelog_contents="$(awk "/$release_version_regex/,/$current_version_regex/" "$changeLogFPath")"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:4509]: Approved

File: MerlinAU.sh [hunk 14]
```diff
@@ -2924,25 +2918,26 @@ _GetLatestFWUpdateVersionFromGithub_()
...
     # Check if a URL was found
-    if [ -z "$download_url" ]; then
+    if [ -z "$download_url" ]
+    then
         echo "**ERROR** **NO_GITHUB_URL**"
         return 1
     else
         # Extract the version from the download URL or release data
-        local version=$(echo "$download_url" | grep -oE "${PRODUCT_ID}[_-][0-9.]+[^/]*" | sed "s/${PRODUCT_ID}[_-]//;s/.w$//;s/_/./g")
+        local version="$(echo "$download_url" | grep -oE "${PRODUCT_ID}[_-][0-9.]+[^/]*" | sed "s/${PRODUCT_ID}[_-]//;s/.w$//;s/_/./g")"
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:2940]: Approved

File: MerlinAU.sh [hunk 15]
```diff
@@ -2924,25 +2918,26 @@ _GetLatestFWUpdateVersionFromGithub_()
...
+    # Fetch the latest release data from GitHub #
+    local release_data="$(curl -s "$url")"
 
-    # Construct the grep pattern based on search_type
+    # Construct the grep pattern based on search_type #
     local grep_pattern="\"browser_download_url\": \".*${PRODUCT_ID}.*\(${search_type}\).*\.\(w\|pkgtb\)\""
 
     # Filter the JSON for the desired firmware using grep and head to fetch the URL
-    local download_url=$(echo "$release_data" | 
+    local download_url="$(echo "$release_data" |
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:2928]: Approved

File: MerlinAU.sh [hunk 16]
```diff
@@ -2924,25 +2918,26 @@ _GetLatestFWUpdateVersionFromGithub_()
         search_type="pure\|squashfs\|ubi"
     fi
 
-    # Fetch the latest release data from GitHub
-    local release_data=$(curl -s "$url")
+    # Fetch the latest release data from GitHub #
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:2921]: Approved

File: MerlinAU.sh [hunk 17]
```diff
@@ -2106,14 +2102,12 @@ _DoCleanUp_()
    rm -f "${FW_ZIP_DIR:?}"/*
    "$delBINfiles" && rm -f "${FW_BIN_DIR:?}"/*
 
-   # Move files back to their original location if they were moved
-   if "$moveZIPback"; then
-       mv -f "${FW_ZIP_BASE_DIR}/${ScriptDirNameD}/${FW_FileName}.zip" "$FW_ZIP_FPATH"
-   fi
+   # Move files back to their original location if needed #
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:2105]: Approved

File: MerlinAU.sh [hunk 18]
```diff
@@ -2106,14 +2102,12 @@ _DoCleanUp_()
...
+   "$moveZIPback" && \
+   mv -f "${FW_ZIP_BASE_DIR}/${ScriptDirNameD}/${FW_FileName}.zip" "$FW_ZIP_FPATH"
 
-   if "$moveWback"; then
-       mv -f "${FW_ZIP_BASE_DIR}/${ScriptDirNameD}/${FW_FileName}.${extension}" "$FW_DL_FPATH"
-   fi
+   "$moveWback" && \
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:2109]: Approved

File: MerlinAU.sh [hunk 19]
```diff
@@ -2090,14 +2088,12 @@ _DoCleanUp_()
...
-   local moveZIPback=false
-   local moveWback=false
+   # Check existence of files and preserve based on flags #
+   local moveZIPback=false  moveWback=false
 
    # Move file temporarily to save it from deletion #
-   if "$keepZIPfile" && [ -f "$FW_ZIP_FPATH" ]; then
-       mv -f "$FW_ZIP_FPATH" "${FW_ZIP_BASE_DIR}/$ScriptDirNameD" && moveZIPback=true
-   fi
+   "$keepZIPfile" && [ -f "$FW_ZIP_FPATH" ] && \
```
Comment (ExtremeFiretop) [inline MerlinAU.sh:2095]: Approved

File: MerlinAU.sh [hunk 20]
```diff
@@ -1328,11 +1324,15 @@ _Set_FW_UpdateZIP_DirectoryPath_()
            return 1
        fi
        # Remove now the obsolete directory path #
-       rm -fr "${FW_LOG_DIR:?}"
+       rm -fr "${FW_ZIP_DIR:?}"
```
[the remark this answers is not in the document]
Comment (ExtremeFiretop) [reply MerlinAU.sh:1327]: Approved. Good catch!

**Comment by ExtremeFiretop** [comment]
I just realized we opened over a hundred PRs between this one and the original Gnuton PR hahaha

**Comment by ExtremeFiretop** [comment]
@Martinski4GitHub 

This is so clean. So nice. So pretty to look at.

I noticed a lot of it was just clarifying the logic already there, I'm curious, does that calm the warming in the linter tool? Or do they continue to exist?

Would it help if I just generally followed more closely to these guidelines? (Even though the logic is more or less the same?)

Also. I noticed there are times you expanded the if this then that condition, other times you shrink/contract the statements. Can you help me understand when is best suited for what?

Feel free to flag some examples in the code (for study and learning on my part)

**Comment by ExtremeFiretop** [comment]
Also does this mean we are now at the regression testing stage?

**Comment by Martinski4GitHub** [comment]
> I just realized we opened over a hundred PRs between this one and the original Gnuton PR hahaha
>
You're right; I had not noticed that, LOL!! The PR was opened a little over 4 months ago!!

**Comment by Martinski4GitHub** [comment]
> Also does this mean we are now at the regression testing stage?
>
Yes, absolutely. Regression testing can begin anytime now.

**Comment by Martinski4GitHub** [comment]
> @Martinski4GitHub
> 
> This is so clean. So nice. So pretty to look at.
> 
> I noticed a lot of it was just clarifying the logic already there, I'm curious, does that calm the warning in the linter tool? Or do they continue to exist?
> 
> Would it help if I just generally followed more closely to these guidelines? (Even though the logic is more or less the same?)
> 
>
All "unquoted" variable assignment statements are flagged as warnings by the Linter tool due to non-compliance with coding best practices. Fixing those lines removed about 18 or so warnings so it’s best to follow the rule unless there’s a very specific reason not to, in which case the warning would be ignored. So the vast majority of warnings were in this category.

For example, these are "unquoted" variable assignments:
```
readonly isGNUtonFW=$(_GetFirmwareVariantFromRouter_)
...
...
local release_data=$(curl -s "$url")
```

And these are the same variable assignments but properly "double quoted" now:
```
readonly isGNUtonFW="$(_GetFirmwareVariantFromRouter_)"
...
...
local release_data="$(curl -s "$url")"
```

Other code was modified due to non-compliance with coding standards & best practices.
I'll add the explanation as a code review comment so I can point out the affected code.

File: MerlinAU.sh [hunk 21]
```diff
@@ -5447,19 +5476,21 @@ _RunOfflineUpdateNow_()
...
                 FW_DL_FPATH="${FW_ZIP_DIR}/${FW_FileName}.${extension}"
                 _GnutonBuildSelection_
                 set -- $(_GetLatestFWUpdateVersionFromGithub_ "$FW_GITURL_RELEASE" "$firmware_choice")
+                retCode="$?"
             else
                 set -- $(_GetLatestFWUpdateVersionFromWebsite_ "$FW_SFURL_RELEASE")
+                retCode="$?"
             fi
-            if [ $? -eq 0 ] && [ $# -eq 2 ] && \
+            if [ "$retCode" -eq 0 ] && [ $# -eq 2 ] && \
```
Comment (Martinski4GitHub) [inline MerlinAU.sh:5493]: The code lines **5488**, **5491** & **5493** are examples where code was added/modified due to non-compliance with coding standards & best practices. Essentially, you always want to grab the return code of a function right after the function call **without** any other intervening code.

File: MerlinAU.sh [hunk 22]
```diff
@@ -499,37 +500,24 @@ _ScriptVersionStrToNum_()
...
-      fi
-   fi
+   if false  # Change to true for forcing GNUton flag #
+   then hasGNUtonFW=true ; return 0 ; fi
+   ##FOR TESTING/DEBUG ONLY##
+
+   # Check if installed F/W NVRAM vars contain "gnuton" #
+   if echo "$fwInstalledInnerVers" | grep -iq "gnuton" || \
+      echo "$fwInstalledExtendNum" | grep -iq "gnuton"
+   then hasGNUtonFW=true ; fi
```
Comment (Martinski4GitHub) [inline MerlinAU.sh:518]: The function "`_GetFirmwareVariantFromRouter_()`" was modified to "clean up" the previously convoluted logic and have a simpler logic flow. Also, the code that was intended for "testing & debugging" purposes only should be clearly identified and separated from the rest of the code. This was not flagged by the Linter tool, but it was done to follow a coding standard & best practice.

Comment (ExtremeFiretop) [reply MerlinAU.sh:518]: Thanks for the example!

File: MerlinAU.sh (hunk 21 above)
Comment (ExtremeFiretop) [reply MerlinAU.sh:5493]: Ah yes, I see/understand. Thanks for the example!

**Comment by ExtremeFiretop** [comment]
Thanks for all the examples! I'll take note of all these and make sure to try to remember that to help keep the linter tool cleaner of warnings :)

**Comment by Martinski4GitHub** [comment]
> 
> Also. I noticed there are times you expanded the if this then that condition, other times you shrink/contract the statements. Can you help me understand when which is best suited for what?
> 
> Feel free to flag some examples in the code (for study and learning on my part)
>
WRT the "expansion/contraction" of if-then statements, they’re not due to the Linter tool. It’s a stylistic change done primarily for readability & review purposes (i.e. making the code easier to read for the reviewers). The longer and more elaborate if-then clauses are almost always "expanded" for readability.

For example, the following if-then code is not as easy & quick to read & review:
```
if "$isGNUtonFW"; then printf "\nPlease copy your firmware update file (.w or .pkgtb) using the *original* ZIP filename to this directory:"
else printf "\nPlease copy your firmware ZIP file (using the *original* ZIP filename) to this directory:"; fi
```

When compared with the modified "expanded" version:
```
if "$isGNUtonFW"
then
    printf "\nPlease copy your firmware update file (.w or .pkgtb) using the *original* filename to this directory:"
else
    printf "\nPlease copy your firmware ZIP file (using the *original* ZIP filename) to this directory:"
fi
```

OTOH, when the if-then clauses are short & simple, the lines can simplified by "shrinking" them.
For example, here's the BEFORE:
```
if "$isGNUtonFW"
then
    _ManageChangelogGnuton_ "view"
else
    _ManageChangelogMerlin_ "view"
fi
```

Here's the AFTER:
```
if "$isGNUtonFW"
then _ManageChangelogGnuton_ "view"
else _ManageChangelogMerlin_ "view"
fi
```

Note that there are no strict rules or guidelines to follow for this type of change; it’s a subjective call for what makes a piece of code much clearer for readability & review purposes as well as for future maintainability.

**Comment by ExtremeFiretop** [comment]
> > Also. I noticed there are times you expanded the if this then that condition, other times you shrink/contract the statements. Can you help me understand when which is best suited for what?
> > Feel free to flag some examples in the code (for study and learning on my part)
> 
> WRT the "expansion/contraction" of if-then statements, they’re not due to the Linter tool. It’s a stylistic change done primarily for readability & review purposes (i.e. making the code easier to read for the reviewers). The longer and more elaborate if-then clauses are almost always "expanded" for readability.
> 
> For example, the following if-then code is not as easy & quick to read & review:
> 
> ```
> if "$isGNUtonFW"; then printf "\nPlease copy your firmware update file (.w or .pkgtb) using the *original* ZIP filename to this directory:"
> else printf "\nPlease copy your firmware ZIP file (using the *original* ZIP filename) to this directory:"; fi
> ```
> 
> When compared with the modified "expanded" version:
> 
> ```
> if "$isGNUtonFW"
> then
>     printf "\nPlease copy your firmware update file (.w or .pkgtb) using the *original* filename to this directory:"
> else
>     printf "\nPlease copy your firmware ZIP file (using the *original* ZIP filename) to this directory:"
> fi
> ```
> 
> OTOH, when the if-then clauses are short & simple, the lines can simplified by "shrinking" them. For example, here's the BEFORE:
> 
> ```
> if "$isGNUtonFW"
> then
>     _ManageChangelogGnuton_ "view"
> else
>     _ManageChangelogMerlin_ "view"
> fi
> ```
> 
> Here's the AFTER:
> 
> ```
> if "$isGNUtonFW"
> then _ManageChangelogGnuton_ "view"
> else _ManageChangelogMerlin_ "view"
> fi
> ```
> 
> Note that there are no strict rules or guidelines to follow for this type of change; it’s a subjective call for what makes a piece of code much clearer for readability & review purposes as well as for future maintainability.

I always figured this was a subjective change, but I wanted to get a sense as to what constitutes expanding it vs contracting it.

So know I'll know that if it's "simple" i.e one simply task in the if/else I can contract. But if it's multiple longer strings then I should probably be expanding

**Comment by ExtremeFiretop** [comment]
Sorry for the delay in responses.

I told my girlfriend MerlinAU was ready for regression testing after some 4 months of work.

She asked how long that would take. I said "idk maybe days until we are confident" and she gave me a sad face and asked me to play a few rounds of a video game I bought (overcooked) with her in co-op mode on the couch so I figured it was fair if I was going to be busy for a bit 🤣

**Comment by Martinski4GitHub** [comment]
> >   
> > Note that there are no strict rules or guidelines to follow for this type of change; it’s a subjective call for what makes a piece of code much clearer for readability & review purposes as well as for future maintainability.
> 
> I always figured this was a subjective change, but I wanted to get a sense as to what constitutes expanding it vs contracting it. (In your mind) So I can follow long.
> 
> So know I'll know that if it's "simple" i.e one simply task in the if/else I can contract. But if it's multiple longer strings then I should probably be expanding
>
>
Yes, that's roughly "the rule of thumb" to keep in mind for readability & review purposes.

**Comment by ExtremeFiretop** [comment]
Speaking of coop on the couch, anymore though on Plex? 😉 I'm now well over 1000 movies and TV shows. Currently watching some Americas Got Talent cause it's been a few years hahaha

**Comment by Martinski4GitHub** [comment]
> Sorry for the delay in responses.
> 
> I told my girlfriend MerlinAU was ready for regression testing after some 4 months of work.
> 
> She asked how long that would take. I said "idk maybe days until we are confident" and she gave me a sad face and asked me to play a few rounds of a video game I bought (overcooked) with her in co-op mode on the couch so I figured it was fair if I was going to be busy for a bit 🤣
>
>
I hear you, bud!! And I fully understand the situation. I've gone through that myself many times over the years when I had to work overtime to meet a deadline or came home really late & tired from a long day at work, and my wife ended up eating dinner or watching a movie by herself. I know she understands the responsibilities & commitments I have at work, but OTOH she doesn't always have to like it. I get that, so whenever she announces: "Tonight is going to be movie night." I always tried to be there not only for her but also because I really enjoy watching movies next to her (sometimes she makes funny comments or reactions so it's more enjoyable :>).

Also, many times I've gone "radio silent" for a while here on GitHub because I get "summoned by my higher power." So I'm sure you know what I mean :>).

Anyway, no worries about any delay in replying. I always figure that something has come up that's more important than this hobby project of ours.

**Comment by Martinski4GitHub** [comment]
> Speaking of coop on the couch, anymore thought on Plex? 😉 I'm now well over 1000 movies and TV shows. Currently watching some Americas Got Talent cause it's been a few years hahaha
>
>
I'll ask my wife again about checking out your list of movies available. She gets busy with her work and other personal stuff to take care of as well, so that's likely not at the top of her to-do list. But I'll remind her.

**Comment by ExtremeFiretop** [comment]
> > Speaking of coop on the couch, anymore thought on Plex? 😉 I'm now well over 1000 movies and TV shows. Currently watching some Americas Got Talent cause it's been a few years hahaha
> 
> I'll ask my wife again about checking out your list of movies available. She gets busy with her work and other personal stuff to take care of as well, so that's likely not at the top of her to-do list. But I'll remind her.

Sounds good! No rush, I just figured I'd mention the offer is still open. List of movies is only growing!

I think I'll start testing MerlinAU tomorrow evening, it's starting to get late and I don't want to rush anything. I want to take my time testing any features or flows we may have touched.

I still have to compile a list of my intended tests, but that will be some brain food after a coffee tomorrow morning 🤣

**Comment by Martinski4GitHub** [comment]
> > > Speaking of coop on the couch, anymore thought on Plex? 😉 I'm now well over 1000 movies and TV shows. Currently watching some Americas Got Talent cause it's been a few years hahaha
> > 
> > 
> > I'll ask my wife again about checking out your list of movies available. She gets busy with her work and other personal stuff to take care of as well, so that's likely not at the top of her to-do list. But I'll remind her.
> 
> Sounds good! No rush, I just figured I'd mention the offer is still open. List of movies is only growing!
> 
> I think I'll start testing MerlinAU tomorrow evening, it's starting to get late and I don't want to rush anything. I want to take my time testing any features or flows we may have touched.
> 
> I still have to compile a list of my intended tests, but that will be some brain food after a coffee tomorrow morning 🤣
>
>
Sounds good. I've already started testing the code before I submitted the PR changes to make sure that nothing was obviously broken when doing an RMerlin F/W update. But I haven't gone and tested all the menu options. I might have time tomorrow evening (depending on the workload as the release deadline is near - so far my modules & components are testing well, but I get called to help investigate bugs in other modules to help out).

**Comment by ExtremeFiretop** [comment]
> > > > Speaking of coop on the couch, anymore thought on Plex? 😉 I'm now well over 1000 movies and TV shows. Currently watching some Americas Got Talent cause it's been a few years hahaha
> > > 
> > > 
> > > I'll ask my wife again about checking out your list of movies available. She gets busy with her work and other personal stuff to take care of as well, so that's likely not at the top of her to-do list. But I'll remind her.
> > 
> > 
> > Sounds good! No rush, I just figured I'd mention the offer is still open. List of movies is only growing!
> > I think I'll start testing MerlinAU tomorrow evening, it's starting to get late and I don't want to rush anything. I want to take my time testing any features or flows we may have touched.
> > I still have to compile a list of my intended tests, but that will be some brain food after a coffee tomorrow morning 🤣
> 
> Sounds good. I've already started testing the code before I submitted the PR changes to make sure that nothing was obviously broken when doing an RMerlin F/W update. But I haven't gone and tested all the menu options. I might have time tomorrow evening (depending on the workload as the release deadline is near - so far my modules & components are testing well, but I get called to help investigate bugs in other modules to help out).

Sent you a PM on the forums with a link to my test plan.
Have a goodnight buddy :)

**Comment by Martinski4GitHub** [comment]
> > > > > Speaking of coop on the couch, anymore thought on Plex? 😉 I'm now well over 1000 movies and TV shows. Currently watching some Americas Got Talent cause it's been a few years hahaha
> > > > 
> > > > 
> > > > I'll ask my wife again about checking out your list of movies available. She gets busy with her work and other personal stuff to take care of as well, so that's likely not at the top of her to-do list. But I'll remind her.
> > > 
> > > 
> > > Sounds good! No rush, I just figured I'd mention the offer is still open. List of movies is only growing!
> > > I think I'll start testing MerlinAU tomorrow evening, it's starting to get late and I don't want to rush anything. I want to take my time testing any features or flows we may have touched.
> > > I still have to compile a list of my intended tests, but that will be some brain food after a coffee tomorrow morning 🤣
> > 
> > 
> > Sounds good. I've already started testing the code before I submitted the PR changes to make sure that nothing was obviously broken when doing an RMerlin F/W update. But I haven't gone and tested all the menu options. I might have time tomorrow evening (depending on the workload as the release deadline is near - so far my modules & components are testing well, but I get called to help investigate bugs in other modules to help out).
> 
> Sent you a PM on the forums with a link to my test plan. Have a goodnight buddy :)
>
Looks very detailed & thorough. I like it!! I'll use it as a guide when testing RMerlin F/W updates.
Have a good night's sleep, bud.

**Comment by ExtremeFiretop** [comment]
@Martinski4GitHub 

Updated the rights to the URL so you should be able to edit now :)

**Comment by ExtremeFiretop** [comment]
I've started on Gnuton, feel free to start on the Merlin side!

Goodnight buddy

**Comment by Martinski4GitHub** [comment]
@ExtremeFiretop,

Hey, bud, just to let you know. I have tested the latest development code as much as I could (given my limited free time), and so far I have not run into any issues.

Also, as mentioned before, my wife & I will take a 14-day vacation starting Thursday, Aug-22, so I'll be pretty much offline until we return. I may check my email occasionally but likely won't respond in a timely manner.

Take care bud. We'll talk when I get back.

**Comment by ExtremeFiretop** [comment]
> @ExtremeFiretop,
> 
> Hey, bud, just to let you know. I have tested the latest development code as much as I could (given my limited free time), and so far I have not run into any issues.
> 
> Also, as mentioned before, my wife & I will take a 14-day vacation starting Thursday, Aug-22, so I'll be pretty much offline until we return. I may check my email occasionally but likely won't respond in a timely manner.
> 
> Take care bud. We'll talk when I get back.

Enjoy the trip my friend, I'll send off the release once I have some free time as well to finish the testing as much as possible.
Chat in 14 days! (where are you going btw?)

Hope it's somewhere nice!

**Comment by Martinski4GitHub** [comment]
> > @ExtremeFiretop,
> > Hey, bud, just to let you know. I have tested the latest development code as much as I could (given my limited free time), and so far I have not run into any issues.
> > Also, as mentioned before, my wife & I will take a 14-day vacation starting Thursday, Aug-22, so I'll be pretty much offline until we return. I may check my email occasionally but likely won't respond in a timely manner.
> > Take care bud. We'll talk when I get back.
> 
> Enjoy the trip my friend, I'll send off the release once I have some free time as well to finish the testing as much as possible. Chat in 14 days! (where are you going btw?)
> 
> Hope it's somewhere nice!
>
We're going to Hawaii. This will be our 2nd time there. The last time was in 2007.

**Comment by ExtremeFiretop** [comment]
> > > @ExtremeFiretop,
> > > Hey, bud, just to let you know. I have tested the latest development code as much as I could (given my limited free time), and so far I have not run into any issues.
> > > Also, as mentioned before, my wife & I will take a 14-day vacation starting Thursday, Aug-22, so I'll be pretty much offline until we return. I may check my email occasionally but likely won't respond in a timely manner.
> > > Take care bud. We'll talk when I get back.
> > 
> > 
> > Enjoy the trip my friend, I'll send off the release once I have some free time as well to finish the testing as much as possible. Chat in 14 days! (where are you going btw?)
> > Hope it's somewhere nice!
> 
> We're going to Hawaii. This will be our 2nd time there. The last time was in 2007.

I wish! Stay safe out there on those tropical islands you! ;) 
Can be lots of storms out there when it's in season, I'm sure you're going to enjoy it!

**Pull request merged**